### PR TITLE
Initial API for IPFS &IPFS-Cluster support

### PR DESCRIPTION
**Pull request by BooniesFX**
1.add IPFS/cluster api support
2.add new payload type:DataFile
3.create new cli cmd to access IPFS API

Signed-off-by: Xiang Fu <fuxiang@onchain.com>

### CODE REVIEW

File: cli/data/data.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		name := c.String("name")
+		if name == "" {
+			rbuf := make([]byte, 4)
+			rand.Read(rbuf)
+			name = "DNA-" + hex.EncodeToString(rbuf)
+		}
+		//create transaction
+		var tx *transaction.Transaction
+
+		wallet := openWallet(c.String("wallet"), []byte(c.String("password")))
```
Comment (dreamfly281) [inline cli/data/data.go]: It does't work now, as the wallet generated by a global defined password and will be generated by the passwork that user input

Comment (BooniesFX) [inline cli/data/data.go]: this wallet is opened by nodectl  and used to sign transaction, and the passwd is created by user

Comment (qingche123) [inline cli/data/data.go]: test passed

Comment (BooniesFX) [inline cli/data/data.go]: @dreamfly281 c.String("wallet")c.String("password") mean parse the string value after the arg "wallet" and "password" and pass them to openwallet

**Pull request merged**